src/clinical/donor-repo.ts

```diff
@@ -413,27 +416,44 @@ export const donorDao: DonorRepository = {
 		return iterateAllByProgramId(programId);
 	},
 
-	async update(donor: DeepReadonly<Donor>) {
-		const newDonor = new DonorModel(donor);
+	async update(donor: Donor) {
+		const dnr = await setEntityIds(donor);
+		const newDonor = new DonorModel(await setEntityIds(donor));
 		unsetIsNewFlagForUpdate(newDonor);
-
+		console.log('newDonor._id' + newDonor._id);
```

Comment:
There are a lot of console.logs added and I'm not sure which are intentional and which are left over from testing

It would be good to replace those you want to keep with the `L` logger used elsewhere in the app; do a search for `const L` or `L.info` for examples

Reply:
removed all console logs.